library/core/src/convert/mod.rs

```diff
@@ -22,6 +22,15 @@
 //! may be necessary to implement [`Into`] or [`TryInto`] directly when converting to a type
 //! outside the current crate.
 //!
+//! # When to implement these traits
+//!
+//! Generally, the conversion traits in this module are intended to convert between types that
+//! are semantically equivalent and differ only in representation.
```

Comment:
We talked about this in today's @rust-lang/libs-api meeting. We felt that the text below was helpful, but this general description didn't quite work. "semantically equivalent" doesn't feel like it matches some cases in the wild.

For instance, it's fine to have `From`/`TryFrom` implementations between subset/superset types, such as between `Ipv4Addr` and `IpAddr`.

We felt like this needs a different summary of when to use `From`. We weren't quite sure what that summary should be, just that "semantically equivalent and differ only in representation" didn't feel accurate and didn't seem sufficient to capture some cases.